packages/remix/src/utils/instrumentServer.ts

```diff
@@ -262,8 +272,18 @@ function getTraceAndBaggage(): { sentryTrace?: string; sentryBaggage?: string }
 function makeWrappedRootLoader(origLoader: DataFunction): DataFunction {
   return async function (this: unknown, args: DataFunctionArgs): Promise<Response | AppData> {
     const res = await origLoader.call(this, args);
+    const traceAndBaggage = getTraceAndBaggage();
+
+    // Note: `redirect` and `catch` responses do not have bodies to extract
+    if (isResponse(res) && !isRedirectResponse(res) && !isCatchResponse(res)) {
+      const data = await extractData(res);
+
+      if (typeof data === 'object') {
```

Comment:
@AbhiPrasad, the `else` case of this is a bit weird (when data is primitive). It doesn't sound right to return a primitive from a `loader`, but TS definitions allow it.

So, would it make sense if we skip injection in such cases? It may mess up the primitive when we spread it?

Reply:
Yeah I would rather we skip injection here.